### PR TITLE
CoreFX import for ProtectedData type (Windows part)

### PR DESCRIPTION
**Pull request by MaximLipnin**
As part of #7589 

- CoreFX import only for Windows platform (no support for non-Windows platforms, see https://github.com/dotnet/corefx/issues/22510)
- left Mono managed implementation for non-Windows platforms
- removed DataProtectionScope parameter value check (no such a check in CoreFX and NET Framework, see https://github.com/dotnet/corefx/pull/30726) and couple of related unit tests
- added CoreFX xunit tests

### CODE REVIEW

**Comment by filipnavara** [comment]
Is there still any point in having NativeDapiProtection.cs?

**Comment by MaximLipnin** [comment]
no, I will delete it, thanks

File: mcs/class/System.Security/win32_System.Security.dll.sources [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include System.Security.dll.sources
```
Comment (marek-safar) [inline mcs/class/System.Security/win32_System.Security.dll.sources]: Please update also winaot profile

**Comment by MaximLipnin** [comment]
Seems that xunit tests for ProtectedData need to be extracted to some separate test profiles for win32/winaot.

**Comment by MaximLipnin** [comment]
At the moment I don't know what exactly test profile naming should be used.

**Comment by MaximLipnin** [comment]
https://github.com/mono/mono/issues/9395

**Comment by filipnavara** [comment]
At least on macOS the tests that failed show an actual problem with the managed implementation. I don't necessarily think it has to be solved within this PR, but there should probably be an issue tracking it. The path that the functions try to access on macOS is protected and read-only, maybe "/usr/local" instead of "/usr" would work.

**Comment by marek-safar** [comment]
@monojenkins commit csproj

**Pull request merged**